Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs

```
@@ -57,6 +57,11 @@ public ResultCode PopLaunchParameter(ServiceCtx context)
                     throw new NotImplementedException($"Unknown LaunchParameterKind {kind}");
```

After a quick RE, we should throw this exception.
This check is made at the beginning of the function instead and we should add it.
```csharp
if ((kind - 1) > 2 )
{
    return ResultCode.ObjectInvalid;
}
```

---

```cs
if (kind != LaunchParameterKind.UserChannel && kind != LaunchParameterKind.PreselectedUser)
{
    return ResultCode.ObjectInvalid;
}
```
is much more readable.

---

Yeah "we shouldn't" sorry for the mistake! I've just given the RE pseudocode :D

---

Are you sure its doing a `>` comparison? If that's the case then that means there's one kind missing from the enum (with value 3).

---

The value 3 is something but for now it seems reserved/unused.

https://github.com/Ryujinx/Ryujinx/blob/6c9565693fd87ae1af81ed63b5fbdde2a5dbecb8/Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/Types/LaunchParameterKind.cs#L3-L8

---

Can't I just return `ResultCode.ObjectInvalid` from `default:` in the switch statement? ...and add a case for unknown that throws.

---

Yeah it's fine too ;)

---

Added handling for this.